### PR TITLE
refactor: add typing + docstrings for user permissions

### PR DESCRIPTION
**Pull request by jasont7**
## Compatiblity

Any changes that require a newer version of the API are considered breaking
changes.

Common breaking changes:
 - Making a request to a new API endpoint that will return `403` or `404` in older versions of the API.
 - No longer sending request fields required by previous versions of the API 
 - Requiring new fields from an API endpoint that will not be present in
   responses from older versions of the API
 - Breaking backwards compatibility with old response data shapes

Choose one:
 - [x] The changes do not break compatibility
 - [ ] The changes potentially break compatibility

## Pre-Review Checklist
 - [x] All changes are tested
 - [x] All touched code documentation is updated
 - [x] All tests pass in your local environment
 - [x] Deepsource issues have been reviewed and addressed
 - [x] Debugging logging and commented out code has been removed

 
## Screenshots:
_Only required for visual changes_.

### CODE REVIEW

File: src/js/users/components/Permission.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    margin-right: 15px;
+`;
+
+const StyledPermissionItem = styled(BoxGroupSection)`
+    align-items: center;
+    display: flex;
+`;
+
+/**
+ * Permission item with an icon indicating its status.
```
Comment (raastinn) [inline src/js/users/components/Permission.tsx]:
```suggestion
A condensed permission item for use in a list of user permissions
```

File: src/js/users/components/Permission.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * @prop permission - name of the permission.
+ * @prop value - indicates if the permission is granted.
+ */
+export function PermissionItem({ permission, value }: { permission: string; value: boolean }) {
```
Comment (raastinn) [inline src/js/users/components/Permission.tsx]: To keep it more consistent with the other components, define a type  `PermissionItemProps` with the permission item types and doc strings.

File: src/js/users/components/Permission.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    /* Name of the permission */
+    permission: string;
+    /* Indicates if the permission is granted */
+    value: boolean;
+};
+
+/**
+ * Permission item with an icon indicating its status.
+ * @prop permission - name of the permission.
+ * @prop value - indicates if the permission is granted.
```
Comment (ReeceHoffmann) [inline src/js/users/components/Permission.tsx]: I don't think `@props` is part of the tsdoc standard. Also information here is redundant with the information the the type

File: src/js/users/components/Permission.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+type PermissionItemProps = {
+    /* Name of the permission */
+    permission: string;
+    /* Indicates if the permission is granted */
+    value: boolean;
+};
+
+/**
+ * Permission item with an icon indicating its status.
```
Comment (raastinn) [inline src/js/users/components/Permission.tsx]:
```suggestion
 * A condensed permission item for use in a list of user permissions
```

Not sure why but the commit on this didnt apply

**Review by ReeceHoffmann — APPROVED** [review]

**Pull request merged**